packages/components/src/slot-fill/bubbles-virtually/slot.js

```diff
@@ -19,19 +19,22 @@ function Slot(
 	{ name, fillProps = {}, as: Component = 'div', ...props },
 	forwardedRef
 ) {
-	const registry = useContext( SlotFillContext );
+	const { registerSlot, unregisterSlot, ...registry } =
+		useContext( SlotFillContext );
 	const ref = useRef();
 
+	// We want don't fillProps in the deps of the layout effect below because we don't want to
+	// unregister and register the slot whenever fillProps change. Doing so would
+	// cause the fill to be re-mounted, and we are only considering the initial value
+	// of fillProps. Instead, we store that initial value in a ref, but don't update it.
+	const fillPropsRef = useRef( fillProps );
+
 	useLayoutEffect( () => {
-		registry.registerSlot( name, ref, fillProps );
+		registerSlot( name, ref, fillPropsRef.current );
```

Comment:
If the original intention was to "only consider the initial value of `fillProps`", then using a ref may cause that behaviour to change

Reply:
Can you explain that a bit more for me? My hope was that if we store the initial value in the ref, and then don't update that ref at all, then the initial value should be preserved, regardless of what happens to `fillProps` itself later on. Am I not thinking about the ref behavior correctly?

Reply:
Excuses for not explaining myself correctly before. What you say is true, but your code changes may introduce a difference in behavior.

Imagine:

- component renders for the first time, `useLayoutEffect` runs and calls `registerSlot` with the initial value of `fillProps`
- later, the component re-renders and one of the dependencies of this `useLayoutEffect` has changed — this will cause the `useLayoutEffect` to run again and call `registerSlot` again.
  - with the code written as it is on `trunk` , `registerSlot` will use the _latest_ value of `fillProps`, which may be different from the _initial_ value
  - with the changes from this PR, `registerSlot` will keep using the _initial_ value of `fillProps`. This means that if in the meantime the value of `fillProps` changed, it would be ignored.


I'm not exactly sure which one was the intended behaviour here, but I wanted to flag this potential change. @youknowriad , could you advise on which of these two scenarios sounds the most correct to you, please? Thank you!

Reply:
Indeed that's problematic behavior, I missed it. The fillProps used should always be the last ones, not the initial ones but there's no need to re-register if they change, because there's a hook right after that update them when they change.

Reply:
Thank you, Riad!

@chad1008 , probably the best thing to do here is also to revert the changes, and simply add an `eslint-disable` comment.

Reply:
Ah, thank you for clarifying @ciampo, and for the second look @youknowriad!

I think it was the `and we are only considering the initial value of fillProps` part of the existing comment that through me off, but looking more closely after the above explanation I see exactly what you mean. I'm going to remove that 'initial value' bit from the comment, add an ignore, and link back here.

For any future refactors that are led here by that code comment: unless a more comprehensive refactor is needed, the [Latest Ref pattern](https://epicreact.dev/the-latest-ref-pattern-in-react/) might be of use here. We could store `fillProps` in a ref for use in the `useLayoutEffect`, and keep it up to date as it changes... but the ref wouldn't be a dependency so we wouldn't need to add it to the array. This way when the effect _does_ fire, the ref will contain the latest value, but changes to that value won't trigger anything themselves.